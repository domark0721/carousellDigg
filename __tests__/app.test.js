import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import App from '../src/components/app'

describe('containers: App', () => {
    it('should render the App correctly', () => {
        const mockCallBacks = {
            onAddDigg: () => {},
            onUpvoteDigg: () => {},
            onDownvoteDigg: () => {},
        }

        // empty diggList
        let tree = renderer.create(
            <App
                diggList={[]}
                totalDiggCount={0}
                {...mockCallBacks}
            />,
        ).toJSON()

        // expect to match the snapshot
        expect(tree).toMatchSnapshot()

        // with diggs in diggList
        tree = renderer.create(
            <App
                diggList={[{
                    id: 0,
                    topic: 'digg 1 topic',
                    upvote: 10,
                    downvote: 20,
                }, {
                    id: 1,
                    topic: 'digg 2 topic',
                    upvote: 20,
                    downvote: 10,
                }]}
                totalDiggCount={2}
                {...mockCallBacks}
            />,
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should callback when submit new topic', () => {
        const mockCallBack = jest.fn()
        const mockInputValue = 'test topic'
        const elApp = shallow(
            <App
                diggList={[]}
                totalDiggCount={0}
                onAddDigg={mockCallBack}
                onUpvoteDigg={() => {}}
                onDownvoteDigg={() => {}}
            />,
        )

        // trigger submit new topic
        elApp.find('TopicInput').first().props().onSubmitNewTopic(mockInputValue)

        expect(mockCallBack.mock.calls.length).toBe(1)
        expect(mockCallBack.mock.calls[0][0]).toBe(mockInputValue)
    })

    it('should callback when upvote/downvote a topic', () => {
        const mockUpvoteCallBack = jest.fn()
        const mockDownvoteCallBack = jest.fn()

        const elApp = shallow(
            <App
                diggList={[{
                    id: 0,
                    topic: 'digg 1 topic',
                    upvote: 10,
                    downvote: 20,
                }, {
                    id: 1,
                    topic: 'digg 2 topic',
                    upvote: 20,
                    downvote: 10,
                }]}
                totalDiggCount={2}
                onAddDigg={() => {}}
                onUpvoteDigg={mockUpvoteCallBack}
                onDownvoteDigg={mockDownvoteCallBack}
            />,
        )

        // trigger upvote/downvote a topic
        elApp.find('DiggItem').first().props().onVoteDigg(1, 1, 'UP')
        elApp.find('DiggItem').first().props().onVoteDigg(1, 1, 'DOWN')
        elApp.find('DiggItem').first().props().onVoteDigg(1, 1, 'invalid vote type')

        expect(mockUpvoteCallBack.mock.calls.length).toBe(1)
        expect(mockUpvoteCallBack.mock.calls[0][0]).toBe(1)

        expect(mockDownvoteCallBack.mock.calls.length).toBe(1)
        expect(mockDownvoteCallBack.mock.calls[0][0]).toBe(1)
    })
})
