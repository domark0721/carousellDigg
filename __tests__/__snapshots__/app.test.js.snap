// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers: App should render the App correctly 1`] = `
<div
  className="content-wrap"
>
  <h1
    className="title"
  >
    Carousell Digg
  </h1>
  <div
    className="author"
  >
    by Mike Liu
  </div>
  <div
    className="topic-input"
  >
    <h3
      className="input-label"
    >
      Write a new topic
    </h3>
    <textarea
      className="input-textarea"
      onChange={[Function]}
      rows="4"
      value=""
    />
    <div
      className="input-wordcount"
    >
      0
      /255
    </div>
    <div
      className="submit-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      Submit
    </div>
  </div>
  <div
    className="digg-list"
  >
    <div
      className="paging-info"
    >
      total 
      0
       diggs, 
      write some topic now!
    </div>
  </div>
</div>
`;

exports[`containers: App should render the App correctly 2`] = `
<div
  className="content-wrap"
>
  <h1
    className="title"
  >
    Carousell Digg
  </h1>
  <div
    className="author"
  >
    by Mike Liu
  </div>
  <div
    className="topic-input"
  >
    <h3
      className="input-label"
    >
      Write a new topic
    </h3>
    <textarea
      className="input-textarea"
      onChange={[Function]}
      rows="4"
      value=""
    />
    <div
      className="input-wordcount"
    >
      0
      /255
    </div>
    <div
      className="submit-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      Submit
    </div>
  </div>
  <div
    className="digg-list"
  >
    <div
      className="paging-info"
    >
      total 
      2
       diggs, 
      showing top 2
    </div>
    <div
      className="digg-item"
    >
      <div
        className="digg-id"
      >
        #
        0
      </div>
      <p
        className="digg-topic"
      >
        digg 1 topic
      </p>
      <div
        className="digg-buttons-wrap"
      >
        <div
          className="digg-button upvote-button"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="button-label"
          >
            Upvote
          </span>
          <span
            className="button-count"
          >
            10
          </span>
        </div>
        <div
          className="digg-button downvote-button"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="button-label"
          >
            Downvote
          </span>
          <span
            className="button-count"
          >
            20
          </span>
        </div>
      </div>
    </div>
    <div
      className="digg-item"
    >
      <div
        className="digg-id"
      >
        #
        1
      </div>
      <p
        className="digg-topic"
      >
        digg 2 topic
      </p>
      <div
        className="digg-buttons-wrap"
      >
        <div
          className="digg-button upvote-button"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="button-label"
          >
            Upvote
          </span>
          <span
            className="button-count"
          >
            20
          </span>
        </div>
        <div
          className="digg-button downvote-button"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="button-label"
          >
            Downvote
          </span>
          <span
            className="button-count"
          >
            10
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
